src/engine/ruleset/decoders/fortinet/fortinet-fortimail.yml

```diff
@@ -0,0 +1,170 @@
+    # date=2019-12-14 time=07:24:31 device_id=int log_id=oremagn log_part=rnatur type=virus pri=medium from=uptatev to="oditem" src="10.176.31.145" session_id="ineavo" msg=reseo
+    - event.original: >-
+        $DEF_PARSER_HEADER $DEF_PARSER_TYPE_PRI from=<~log.rsa.email.email_src> to="<~log.rsa.email.email_dst>" src="<source.ip>"
+        session_id="<~log.rsa.misc.filter>" msg=<~log.rsa.internal.event_desc>
+    # date=2017-1-6 time=07:11:41 device_id=uaUten log_id=nby log_part=mve type=event subtype=config pri=low user=isau ui=rautodi(10.96.97.81) module=pis submodule=nsequat msg=doloreme
+    - event.original: >-
+        $DEF_PARSER_HEADER $DEF_PARSER_TYPE_SUBTYPE_PRI user=<user.name> ui=<~log.rsa.network.network_service>\(<source.ip>\) module=<~> submodule=<~> msg=<~log.rsa.internal.event_desc>
+    # date=2018-3-25 time=09:31:24 device_id=obeataev log_id=umf log_part=olesti type=event subtype=config pri=low user=quaeabil ui=emip module=aturQu submodule=itesse msg="iamqui"
+    - event.original: >-
+        $DEF_PARSER_HEADER $DEF_PARSER_TYPE_SUBTYPE_PRI user=<user.name> ui=<~log.rsa.network.network_service> module=<~> submodule=<~> msg="<~log.rsa.internal.event_desc>"
+    # date=2016-6-5 time=21:33:08 device_id=ntutl log_id=caecatc log_part=onsequat type=event subtype=update pri=low msg="edquiano"
+    - event.original: >-
+        $DEF_PARSER_HEADER $DEF_PARSER_TYPE_SUBTYPE_PRI msg="<~log.rsa.internal.event_desc>"
+
+normalize:
+  - map:
+    - event.dataset: fortinet.fortimail
+    - event.module: fortinet
+    - fileset.name: fortimail
+    - input.type: log
+    - observer.product: FortiMail
+    - observer.type: Firewall
+    - observer.vendor: Fortinet
+    - rsa.email.email_dst: $~log.rsa.email.email_dst
+    - rsa.email.email_src: $~log.rsa.email.email_src
+    - rsa.email.subject: $~log.rsa.email.subject
+    - rsa.email.subject: $~log.rsa.misc.event_type
+    - rsa.internal.event_desc: $~log.rsa.internal.event_desc
+    - rsa.internal.messageid: $event.action
+    - rsa.internal.messageid: $~log.rsa.internal.messageid
+    - rsa.misc.checksum: $~log.rsa.misc.checksum
+    - rsa.misc.client: $~log.rsa.misc.client
+    - rsa.misc.context: $~log.rsa.misc.context
+    - rsa.misc.disposition: $~log.rsa.misc.disposition
+    - rsa.misc.event_type: $event.action
+    - rsa.misc.filter: $~log.rsa.misc.filter
+    - rsa.misc.filter: $~log.rsa.misc.filter
+    - rsa.misc.log_session_id: $~log.rsa.misc.filter
+    - rsa.misc.msgIdPart1: $event.action
+    - rsa.misc.msgIdPart2: $~log.rsa.internal.messageid
+    - rsa.misc.reference_id1: $~log.rsa.misc.reference_id1
+    - rsa.misc.reference_id: $event.code
+    - rsa.misc.result: $~log.rsa.misc.result
+    - rsa.misc.severity: $log.level
+    - rsa.misc.virusname: $~log.rsa.misc.virusname
+    - rsa.network.domain: $server.domain
+    - rsa.web.fqdn: $~log.rsa.web.fqdn
```

Comment:
If all the fields are called the same, maybe you could just do an `ef_merge` of the RSA fields instead of copying so many values. Please, check that out.

Reply:
In this case there are few fields, it would be more difficult to eliminate the fields that would not be mapped.